src/Transport/Sending/DefaultBatchedOperationsToBrokeredMessagesConverter.cs

```diff
@@ -63,7 +63,10 @@ void SetReplyToAddress(OutgoingMessage outgoingMessage, BrokeredMessage brokered
         {
             if (outgoingMessage.Headers.ContainsKey(Headers.ReplyToAddress))
             {
-                var replyTo = new EntityAddress(outgoingMessage.Headers[Headers.ReplyToAddress]);
+                var replyToAddress = outgoingMessage.Headers[Headers.ReplyToAddress];
+                if (string.IsNullOrWhiteSpace(replyToAddress)) return;
```

Comment:
the empty header was actually created by the code on the other comment. But nothing says other causes could exist